subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/VariantBackedConfigurationMetadata.java

```diff
@@ -122,6 +130,17 @@ public ComponentArtifactMetadata artifact(IvyArtifactName artifact) {
 
     @Override
     public List<? extends DependencyMetadata> getDependencies() {
-        return dependencies;
+        if (calculatedDependencies == null) {
+            if (dependencyMetadataRules == null) {
+                calculatedDependencies = dependencies;
+            } else {
+                calculatedDependencies = new ArrayList<GradleDependencyMetadata>(dependencies);
+                for (Action<DependenciesMetadata> dependenciesMetadataAction : dependencyMetadataRules.getActions()) {
```

Comment:
❌ This logic seems to belong into the `DependencyMetadataRules` class instead of having the algorithm copied in several places.

Reply:
Sure it does. Totally missed this. Thanks.